src/Microsoft.Extensions.OData.Migration/MigrationOptions.cs

```diff
@@ -0,0 +1,20 @@
+﻿using System;
+
+namespace Microsoft.Extensions.OData.Migration
+{
+    /// <summary>
+    /// Container for parameters to pass into middleware.
+    /// </summary>
+    public sealed class MigrationOptions
+    {
+        // Service root (e.g. the https://foo:80/odata part of https://foo:80/odata/Products) is required for parser
+        public Uri ServiceRoot { get; set; }
+
+        // V4 model to validate request
+        public Microsoft.OData.Edm.IEdmModel V4Model { get; set; }

```

Comment:
This is more than an option -- I wonder if we should be passing this to the UseMigrationMiddleware